autosklearn/ensemble_builder.py

```diff
@@ -692,25 +696,71 @@ def predict(self, set_: str,
 
     def _delete_non_candidate_models(self, candidates):
         candidates = [os.path.split(cand)[1] for cand in candidates]
-        for model_path in self.read_preds.keys():
-            if self.read_preds[model_path]['deleted']:
+        for pred_path in self.read_preds.keys():
+            if self.read_preds[pred_path]['deleted']:
                 continue
-            match = self.model_fn_re.search(model_path)
-            _num_run = int(match.group(2))
+            match = self.model_fn_re.search(pred_path)
+            _seed = match.group(1)
+            _num_run = match.group(2)
+            _budget = match.group(3)
             # Do not remove the dummy prediction!
-            if _num_run == 1:
+            if int(_num_run) == 1:
                 continue
-            model_file = os.path.split(model_path)[1]
-            if model_file not in candidates:
+            pred_file = os.path.split(pred_path)[1]
+            if pred_file not in candidates:
+                # Messages logged bellow should match what is asserted
+                # in test_delete_non_candidate_models(). Please, ensure consistency.
+
+                # Delete prediction file
+                try:
+                    os.remove(pred_path)
+                    self.logger.info(
+                        'Deleted prediction file of non-candidate model %s', pred_path)
+                    self.read_preds[pred_path]['deleted'] = True
+                except Exception as e:
+                    self.logger.error(
+                        'Failed to delete prediction file of non-candidate model %s due'
+                        'to error %s', pred_path, e)
+
+                # Delete model file
+                model_name = '%s.%s.%s.model' % (_seed, _num_run, _budget)
+                model_path = os.path.join(self.dir_models, model_name)
                 try:
                     os.remove(model_path)
-                    self.logger.info("Deleted file of non-candidate model %s", model_path)
-                    self.read_preds[model_path]['deleted'] = True
+                    self.logger.info('Deleted file of non-candidate model %s', model_path)
                 except Exception as e:
                     self.logger.error(
-                        'Failed to delete non-candidate model %s due to error %s',
+                        'Failed to delete file of non-candidate model %s due to error %s',
                         model_path, e)
 
+                # Delete validation prediction file
+                pred_valid_name = 'predictions_valid' + match.group(0)
+                pred_valid_path = os.path.join(self.dir_valid, pred_valid_name)
+                if os.path.exists(pred_valid_path):
```

Comment:
Deleting if exists, though it works, sounds a bit strange to me. But I couldn't find any other flag readily available for this task. A good one would be `if self.read_preds[pred_path][Y_VALID] is not None`.  Nevertheless `get_n_best_preds()` set this value to  `None` for non-candidate models.